Add tvOS support

muralex-e opened the pull request:
In this PR I've tried to add tvOS targets to `moko-resources` project.

As mentioned in #739, the core implementation was already done in Apple generators, so I followed the approach in #648.

I've done the following:

- Added  tvOS targets and source seta to extended convention and Apple main convention.
- Implemented the expected classes and methods for tvOS targets.
- Added missing mocks for tvOS "resources-test" project.

I've tried to build the tvOS target in the sample ios app, but I'm getting the error
`No matching variant of dev.icerock.moko:resources:0.24.1 was found.`

Maybe you can suggest how I can test the tvOS target before it is added in the release version.

muralex-e committed:
add tvOs source sets to extended convention

muralex-e committed:
add tvOS source sets to apple main convention

muralex-e committed:
implemented required expect classes and methods for tvOS

muralex-e committed:
add missing mocks for tvOS to resources-test project

Alex009 commented on resources-build-logic/src/main/kotlin/apple-main-convention.gradle.kts:
```diff
@@ -37,4 +37,10 @@ kotlin {
         this.dependsOn(sourceSets.getByName("appleMain"))
     }
+
+    sourceSets.matching {
+        it.name == "tvosMain"
+    }.configureEach {
+        this.dependsOn(sourceSets.getByName("appleMain"))
+    }
```
better to use 
```
val appleSourceSets = listOf("watchos", "tvos")
val appleMainSourceSets = appleSourceSets.map { "${it}Main" }
val appleTestSourceSets = appleSourceSets.map { "${it}Test" }

sourceSets
    .matching { it.name in appleMainSourceSets }
    .configureEach { this.dependsOn(sourceSets.getByName("appleMain")) }
    
sourceSets
    .matching { it.name in appleTestSourceSets }
    .configureEach { this.dependsOn(sourceSets.getByName("appleTest")) }
```

Alex009 commented:
>I've tried to build the tvOS target in the sample ios app, but I'm getting the error
>No matching variant of dev.icerock.moko:resources:0.24.1 was found.
>
>Maybe you can suggest how I can test the tvOS target before it is added in the release version.

you should up version in `gradle/moko.versions.toml` and run `./gradlew publishToMavenLocal`

muralex-e committed:
refactoring for watchOS and tvoOS source sets in apple main convention

muralex-e commented:
I have added the requested changes. Thank you for the suggestion, it helped with the error I had. 
However, I still can’t compile the tvOS project with the mokoresources.

So, basically I tried to follow the commits from PR #648 and added a tvOS target in the samples/kotlin-ios-app. TvOS app can be built without moko resources, but when I try to import mpp-library I’m getting the error:

`No such module 'MppLibrary'`

I also tried creating a new tvOS application in the project, but I encountered the same error and couldn't figure out how to fix it.

Would it be useful to add here the commits with the tvOS target in the existing app or the newly created app that I tried, even though it is not working properly?

Alex009 commented:
>Would it be useful to add here the commits with the tvOS target in the existing app or the newly created app that I tried, even though it is not working properly?

needed to add `tvos-app` in `resources-gallery` sample, and to `compose-resources-gallery` if tvos works with compose multiplatform

muralex-e committed:
fixed ImageResource implementation for tvOS

muralex-e committed:
added tvOS targets and sourcesets to mpp-librarary in resource-gallery

muralex-e committed:
created tvOS app in the resources gallery
